### PR TITLE
lib/9pfs: Add README.md file for 9pfs

### PR DESCRIPTION
**Pull request by DeliaPavel**
### Prerequisite checklist

 - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [ ] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]

### CODE REVIEW

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
This filesystem uses the [9P network protocol](https://en.wikipedia.org/wiki/9P_(protocol)) for communication between the two instances.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]: Add the structures with the updated comments from your pr #794.

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* The `dev` field which refers to a 9P device and is initialized in [`uk_9p_dev_connect`](https://github.com/unikraft/unikraft/blob/staging/lib/uk9p/9pdev.c#L237)
```
Use one sentence per line and one line per sentence.

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* The `trans` field describing the wanted transport to use (such as `virtio`, `unix`, `tcp` etc.).
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
  It can be `UK_9P_PROTO_2000U`, `UK_9P_PROTO_2000L` or `UK_9P_PROTO_MAX`.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
The `uk_9pfs_mount_data` structure is used for operations that can be done on the files in the filesystem (e.g `open`, `close`, `readdir`).
It contains:
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* The `fid`, an unique file identifier which identifies the file handle.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
  In the `uk_9pfs_mount_data` we can also find the `readdir_sz` which is the size of data in the `readdir_buf`
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
   $ mkdir fs0
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
$ qemu-system-x86_64 -fsdev local,id=myid,path=./fs0/,security_model=none -device virtio-9p-pci,fsdev=myid,mount_tag=fs0 -kernel build/06-adding-filesystems_kvm-x86_64 -nographic
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
The complete command for running this program with `qemu` is:
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* `-device virtio-9p-pci,fsdev=myid,mount_tag=fs0` - create a device with the 9pfs type, assign the `myid` for the `-fsdev` option and also assign the mount tag that we configured above (`fs0`).
  Unikraft will look after that mount tag when trying to mount the filesystem, so it is important that the mount tag from the configuration is the same as the one given as argument to qemu.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* `-kernel build/app-helloworld_kvm-x86_64` - tells QEMU that it will run a kernel; if this parameter is omitted, QEMU will think it runs a raw file.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* `-nographic` - prints the output of QEMU to the standard output, it doesn’t open a graphical window.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
1. Under `vfscore: Configuration  --->` select `Automatically mount a root filesystem (/)`.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
1. Select `Default root filesystem -> 9pfs`.
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Thank you @DeliaPavel, see the few comments.

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* The `uname` field, which refers to the user name attempting the connection.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* The `aname` specifing the file system name to mount.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md:14]: Align the structure members like they are done in the `uk_9pfs_file_data` and `uk_9pfs_node_data` below.

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* The `readdir_buf`, which contains the information read during the `readdir()` calls.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* The `readdir_off`, which refers to the offset from within the `readdir_buf` where we can find information about the next child.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
1. Under `Library Configuration`, select `vfscore:` `VFS Core Interface`.
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
1. Set the `Default root device` option to `fs0` (instead of the default `rootfs`).
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
1. We want to run Unikraft with QEMU / KVM, so we must select KVM guest in the `Platform Configuration` menu. 
For `9pfs` we also need to enable, in the KVM guest options menu, `Virtio --->`, then `Virtio PCI device support`.
```
Use one sentence per line.

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
* `-kernel build/app-helloworld_kvm-x86_64` - tells QEMU that it will run a kernel image; if this parameter is omitted, QEMU will think it runs a raw file.
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few more comments @DeliaPavel.

**Review by razvand — CHANGES_REQUESTED** [review]
Hi, @DeliaPavel, this looks very good. Please remove the reference to `fs0`. Just leave `rootfs`. `fs0` is a rather stupid leftover from the `qemu-guest` script that we should fully discard from the documentation and scripts.

**Review by razvand — COMMENTED** [review]
All good with me. After @StefanJum approves it, I will approve it myself, and have it merged.

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
The `uk_9pfs_mount_data` structure is used for operations that can be done on the files in the filesystem (e.g. `open`, `close`, `readdir`).
```

File: lib/9pfs/README.md
Comment (StefanJum) [inline lib/9pfs/README.md]:
```suggestion
$ qemu-system-x86_64 -fsdev local,id=myid,path=./rootfs/,security_model=none -device virtio-9p-pci,fsdev=myid,mount_tag=rootfs -kernel build/app-helloworld_kvm-x86_64 -nographic
```
Change the image name to the one described below in the command options.

Comment (StefanJum) [reply lib/9pfs/README.md]: You could also break this command in more lines using `\` if you think it looks better. If not, just leave it like this.

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few very minor comments @DeliaPavel, I'll approve it after you solve them.

**Review by StefanJum — COMMENTED** [review]
All good now, thanks @DeliaPavel.
@razvand do your thing.

**Review by StefanJum — APPROVED** [review]
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Thanks, @DeliaPavel.

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**
